common/collection/sync_map.go

```diff
@@ -68,3 +95,17 @@ func (m *SyncMap[K, V]) Pop(key K) (value V, ok bool) {
 	}
 	return value, ok
 }
+
+func (m *SyncMap[K, V]) PopAll() []KeyValuePair[K, V] {
+	m.Lock()
+	defer m.Unlock()
+	var result []KeyValuePair[K, V]
+	for k, v := range m.contents {
+		result = append(result, KeyValuePair[K, V]{
+			Key:   k,
+			Value: v,
+		})
+	}
+	maps.Clear(m.contents)
+	return result
+}
```

Comment:
a simpler/faster approach would be:
```suggestion
func (m *SyncMap[K, V]) PopAll() map[K]V {
	m.Lock()
	defer m.Unlock()
	contents := m.contents
	m.contents = make(map[K]V)
	return contents
}
```
this way is okay too if you want

Reply:
if you do use the slice approach, it's faster to do `clear(m.contents)` (new in go 1.21) than to delete each key individually

Reply:
Thank you. Didn't know it was there.